src/artemis/experiment_plans/tests/conftest.py

```diff
@@ -191,3 +238,203 @@ def mock_rotation_subscriptions(test_rotation_params):
     ):
         subscriptions = RotationCallbackCollection.from_params(test_rotation_params)
     return subscriptions
+
+
+def fake_read(obj, initial_positions, _):
+    initial_positions[obj] = 0
+    yield Msg("null", obj)
+
+
+def do_rotation_plan_for_tests(
+    run_engine,
+    callbacks,
+    sim_und,
+    sim_synch,
+    sim_slits,
+    sim_flux,
+    sim_att,
+    expt_params,
+    sim_sgon,
+    sim_zeb,
+    sim_bl,
+    sim_det,
+):
+    with (
+        patch(
+            "bluesky.preprocessors.__read_and_stash_a_motor",
+            fake_read,
+        ),
+        patch(
+            "artemis.experiment_plans.rotation_scan_plan.RotationCallbackCollection.from_params",
+            lambda _: callbacks,
+        ),
+        patch("dodal.beamlines.i03.undulator", lambda: sim_und),
+        patch("dodal.beamlines.i03.synchrotron", lambda: sim_synch),
+        patch("dodal.beamlines.i03.s4_slit_gaps", lambda: sim_slits),
+        patch("dodal.beamlines.i03.flux", lambda: sim_flux),
+        patch("dodal.beamlines.i03.attenuator", lambda: sim_att),
+    ):
+        run_engine(
+            rotation_scan_plan(
+                expt_params,
+                sim_sgon,
+                sim_zeb,
+                sim_bl,
+                sim_att,
+                sim_det,
+            )
+        )
+
+
+def setup_and_run_rotation_plan_for_tests(
```

Comment:
Should: I think because this fixture is specific to rotation scans it should go in `test_rotation_scan_plan` rather than `conftest`